extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package io.quarkus.vault;
```

Comment:
Is this class complete (there is no support for the user pass...) ?
Another question: Why `VaultService` can not also do what 'VaultProxy' does ?

Reply:
there are 2 levels in the api:
 - VaultProxy: simple rest client
 - VaultService: handles transparently auth (initial, extension and recreate) and 2 main services: reading static secrets (kv version 1 and 2) and generating dynamic db credentials (initial lease creation, extension and recreate)

In VaultService we handle the login transparently based on your configuration. so there is no loginXXX but just getToken()